### PR TITLE
feat: add Actor Standby documentation

### PR DESCRIPTION
**Pull request by mhamas**
CleanShot 2024-06-27 at 22 07 55@2x

### CODE REVIEW

File: sources/platform/actors/running/actor_standby.md
Comment (jirimoravcik) [inline sources/platform/actors/running/actor_standby.md:42]: build is missing here

File: sources/platform/actors/running/actor_standby.md
Comment (jirimoravcik) [inline sources/platform/actors/running/actor_standby.md:56]: This one seems to have "build", "memoryMbytes", "desiredRequestsPerActorRun" missing, I guess some old design?

**Review by jirimoravcik — APPROVED** [review]

File: sources/platform/actors/running/actor_standby.md
Comment (fnesveda) [inline sources/platform/actors/running/actor_standby.md:2]: Are we using "Actor standby" or "Actor Standby"? I vote for Standby.
```suggestion
title: Actor Standby
```

File: sources/platform/actors/running/actor_standby.md
Comment (fnesveda) [inline sources/platform/actors/running/actor_standby.md:14]:
```suggestion
However, in some applications, waiting for an Actor to start is not an option. Actor Standby mode solves this problem by enabling you to have the Actor ready
```

File: sources/platform/actors/running/actor_standby.md
Comment (fnesveda) [reply sources/platform/actors/running/actor_standby.md:56]: Yeah, we'll have to update the screenshots once the UI is actually merged in the Console.

File: sources/platform/actors/running/actor_standby.md
Comment (fnesveda) [inline sources/platform/actors/running/actor_standby.md:49]: I would put a note here that the Standby Actors might incur unexpected costs because they're running in the background, waiting for requests.
```suggestion
No, the standby runs are billed in the same fashion as the normal runs.
However, there might be unexpected costs for running Actors in the Standby mode, as the Actors are running in the background and consuming resources even when no requests are being sent, until they are terminated after the idle timeout period.
```

File: sources/platform/actors/running/actor_standby.md
Comment (fnesveda) [reply sources/platform/actors/running/actor_standby.md:42]:
```suggestion
- **Build** - The Actor build that the runs of the Standby Actor will use. Can be either a build tag (e.g. `latest.`), or a build number (e.g. `0.1.2`).

```

**Review by fnesveda — CHANGES_REQUESTED** [review]
Looks good, I left a few comments.

@jirimoravcik since Maťo is on vacation, can you check my suggestions, and if you're fine with them, just commit them?

File: sources/platform/actors/running/actor_standby.md
Comment (stetizu1) [inline sources/platform/actors/running/actor_standby.md:30]:
```suggestion
it well. Please head to the Actor README to learn more about the capabilities of your chosen Actor.
```

File: sources/platform/actors/running/actor_standby.md
Comment (stetizu1) [inline sources/platform/actors/running/actor_standby.md:34]:
```suggestion
the system starts new Actor runs, which you will see in the Actor runs tab, with the origin set to Standby.
```

File: sources/platform/actors/running/actor_standby.md
Comment (stetizu1) [inline sources/platform/actors/running/actor_standby.md:38]:
```suggestion
- **Max requests per run** - The maximum number of concurrent HTTP requests a single standby Actor run can accept. If this limit is exceeded, the system starts a new Actor run to handle the request, which may take a few seconds.
```

File: sources/platform/actors/running/actor_standby.md
Comment (stetizu1) [inline sources/platform/actors/running/actor_standby.md:39]:
```suggestion
- **Desired requests per run** - The number of concurrent HTTP requests a single Standby Actor run is configured to handle. If this limit is exceeded, the system preemptively starts a new Actor run to handle the additional requests.
```

File: sources/platform/actors/running/actor_standby.md
Comment (stetizu1) [inline sources/platform/actors/running/actor_standby.md:40]:
```suggestion
- **Memory (MB)** - The amount of memory (RAM) allocated for the Actor in Standby mode, in megabytes. With more memory, the Actor can typically handle more requests in parallel, but this also increases the number of compute units consumed and the associated cost.
```

File: sources/platform/actors/running/actor_standby.md
Comment (stetizu1) [inline sources/platform/actors/running/actor_standby.md:41]:
```suggestion
- **Idle timeout (seconds)** - If a Standby Actor run doesn’t receive any HTTP requests within this time, the system will terminate the run. When a new request arrives, the system might need to start a new Standby Actor run to handle it, which can take a few seconds. A higher idle timeout improves responsiveness but increases costs, as the Actor remains active for a longer period.
```

File: sources/platform/actors/running/actor_standby.md
Comment (stetizu1) [inline sources/platform/actors/running/actor_standby.md:45]:
```suggestion
You can then head to the Standby tab of the created Task and modify the configuration as needed. Note that the task has a specific hostname, so make
```

File: sources/platform/actors/running/actor_standby.md
Comment (stetizu1) [reply sources/platform/actors/running/actor_standby.md:49]: * => Standby runs ?

File: sources/platform/actors/running/actor_standby.md
Comment (stetizu1) [inline sources/platform/actors/running/actor_standby.md:55]:
```suggestion
You can head to the Settings tab of your Actor, enable Standby mode, and set the default configuration.
```

File: sources/platform/actors/running/actor_standby.md
Comment (stetizu1) [inline sources/platform/actors/running/actor_standby.md:58]:
```suggestion
Please make sure to properly describe your Actors, their endpoints, and the schema for their
```

File: sources/platform/actors/running/actor_standby.md
Comment (stetizu1) [inline sources/platform/actors/running/actor_standby.md:62]: [To consider]: I am thinking that some readme example would be nice, with [#actor-standby header](https://github.com/apify/apify-core/pull/16488#discussion_r1661270287)

**Review by stetizu1 — APPROVED** [review]
This should be probably checked by some native speaker 😅 
Also not sure that mode has "the" article. 🤔 

Content-wise ok, just the comments that are already there + nits

File: sources/platform/actors/running/actor_standby.md
Comment (TheoVasilis) [inline sources/platform/actors/running/actor_standby.md:29]:
```suggestion
not support this mode; the run might fail or return empty results. The normal mode is always supported in Standby Beta, even for Actors that don't handle
```

File: sources/platform/actors/running/actor_standby.md
Comment (TheoVasilis) [inline sources/platform/actors/running/actor_standby.md:13]:
```suggestion
Traditional Actors are designed to run a single task and then stop. They're mostly intended for batch jobs, such as when you need to perform a large scrape or data processing task.
```

File: sources/platform/actors/running/actor_standby.md
Comment (TheoVasilis) [inline sources/platform/actors/running/actor_standby.md:33]:
```suggestion
When you use the Actor in Standby mode, the system automatically scales the Actor to accommodate the incoming requests. Under the hood,
```

File: sources/platform/actors/running/actor_standby.md
Comment (TheoVasilis) [inline sources/platform/actors/running/actor_standby.md:51]:
```suggestion
However, running Actors in Standby mode might have unexpected costs, as the Actors run in the background and consume resources even when no requests are being sent until they are terminated after the idle timeout period.
```

File: sources/platform/actors/running/actor_standby.md
Comment (TheoVasilis) [inline sources/platform/actors/running/actor_standby.md:64]:
```suggestion
No, Standby mode is currently in Beta, and monetization is not supported.
```

**Review by TheoVasilis — CHANGES_REQUESTED** [review]

**Review by TheoVasilis — APPROVED** [review]

**Review by TC-MO — APPROVED** [review]
LGTM

File: sources/platform/actors/running/actor_standby.md
Comment (fnesveda) [inline sources/platform/actors/running/actor_standby.md:63]: This part is completely missing some development instructions. Let's add them once we have the `Actor.config.standbyPort` part in SDK ready.

**Review by fnesveda — CHANGES_REQUESTED** [review]
Looks good, except that we need to add a section about how to actually develop the Standby Actors, and some example Actor etc.

File: sources/platform/actors/running/images/actor_standby/standby-tab.png
Comment (stetizu1) [inline sources/platform/actors/running/images/actor_standby/standby-tab.png:1]: We will need a new screenshot here once more 😅 Sorry. Maybe wait for @adelamikk to approve current core changes

Comment (stetizu1) [reply sources/platform/actors/running/images/actor_standby/standby-tab.png:1]: So probably can be done now.

Comment (jirimoravcik) [reply sources/platform/actors/running/images/actor_standby/standby-tab.png:1]: Done.

File: sources/platform/actors/running/actor_standby.md
Comment (jirimoravcik) [reply sources/platform/actors/running/actor_standby.md]: I added example code

File: sources/platform/actors/running/actor_standby.md
Comment (fnesveda) [inline sources/platform/actors/running/actor_standby.md:96]: I would skip this and go straight for the example with the HTTP server, it's not too complicated, so there's no need to split it into two examples.

File: sources/platform/actors/running/actor_standby.md
Comment (fnesveda) [inline sources/platform/actors/running/actor_standby.md:134]: Can't you just use `Actor.config.standby_port` here?

**Review by fnesveda — CHANGES_REQUESTED** [review]
Looks good, I left a few suggestions

File: sources/platform/actors/running/actor_standby.md
Comment (jirimoravcik) [reply sources/platform/actors/running/actor_standby.md:134]: you're right, but it's not mentioned in docs https://docs.apify.com/sdk/python/reference/class/Configuration

Comment (fnesveda) [reply sources/platform/actors/running/actor_standby.md:134]: It's there, just not very visible.
Screenshot 2024-07-08 at 15 23 47

It's also mentioned in the `Actor` class docs: https://docs.apify.com/sdk/python/reference/class/Actor#config

But not in the code example in the Configuration & env vars section: https://docs.apify.com/sdk/python/docs/concepts/configuration

Comment (fnesveda) [reply sources/platform/actors/running/actor_standby.md:134]: I've mentioned it in the issue here: https://github.com/apify/apify-sdk-python/issues/147

File: sources/platform/actors/running/actor_standby.md
Comment (jirimoravcik) [reply sources/platform/actors/running/actor_standby.md]: Done

File: sources/platform/actors/running/actor_standby.md
Comment (jirimoravcik) [reply sources/platform/actors/running/actor_standby.md]: Done

File: sources/platform/actors/running/actor_standby.md
Comment (fnesveda) [inline sources/platform/actors/running/actor_standby.md:77]: This way the chosen language will be synced with other tab groups across the documentation
```suggestion
<Tabs groupId="main">
```

File: sources/platform/actors/running/actor_standby.md
Comment (fnesveda) [inline sources/platform/actors/running/actor_standby.md:109]: Let's use single quotes for unification
```suggestion
        with HTTPServer(('', Actor.config.standby_port), GetHandler) as http_server:
```

**Review by fnesveda — APPROVED** [review]
Looks good, just a few last suggestions and then we're good to go

**Review by fnesveda — APPROVED** [review]

**Review by TheoVasilis — APPROVED** [review]

**Pull request merged**
